addons/point_of_sale/models/pos_session.py

```diff
@@ -1750,7 +1750,7 @@ def _get_pos_ui_account_tax(self, params):
             groupby=["tax_id"],
         )
         tax_id_to_factor_sum = {g['tax_id'][0]: g['factor_percent']/100 for g in groups}
-        for tax in taxes:
+        for tax in filter(lambda t: t['amount_type'] != 'group', taxes):
```

Comment:
With this change, we're only preventing the error. If we need to do extra work, what should be the more correct fix? Can't we support taxes with `amount_type = 'group'`?

Reply:
When a tax is `group` it doesn't have "Distribution for invoices" lines. So it can not have `sum_repartition_factor`. I assumed we don't use `sum_repartition_factor` when it's `group`.
We support `amount_type = 'group'` and recursively flatten them with `_collect_taxes` function.
My first solution was to flatten it recursively and compute the `sum_repartition_factor`,  what is your opinion?